File: common/src/main/java/org/valkyrienskies/clockwork/platform/SharedValues.java
package org.valkyrienskies.clockwork.platform;

import com.simibubi.create.foundation.item.render.CustomRenderedItemModelRenderer;
import com.tterrag.registrate.util.entry.EntityEntry;
import dev.architectury.injectables.annotations.ExpectPlatform;
import net.minecraft.world.item.BlockItem;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.Item;
import org.valkyrienskies.clockwork.content.kinetics.sequenced_seat.SequencedSeatEntity;
import org.valkyrienskies.clockwork.platform.api.network.PacketChannel;

import java.util.ArrayList;
import java.util.Set;
import java.util.function.BiConsumer;

public class SharedValues {

    @ExpectPlatform
    public static PacketChannel getPacketChannel() {
        throw new AssertionError();
    }

    @ExpectPlatform
    public static BiConsumer<CWItem, CustomRenderedItemModelRenderer> customRenderedRegisterer() {
        throw new AssertionError();
    }

    @ExpectPlatform
    public static BiConsumer<BlockItem, CustomRenderedItemModelRenderer> customBlockItemRenderedRegisterer() {
        throw new AssertionError();
    }

    //region Entities

    @ExpectPlatform
    public static EntityEntry<SequencedSeatEntity> getSequencedSeat() {
        throw new AssertionError();
    }
    //endregion

    @ExpectPlatform
    public static ArrayList<Item> getFrostingBuckets() {
        throw new AssertionError();
    }
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/platform/SharedValues.kt
package org.valkyrienskies.clockwork.platform

import com.simibubi.create.foundation.item.render.CustomRenderedItemModelRenderer
import com.tterrag.registrate.util.entry.BlockEntityEntry
import com.tterrag.registrate.util.entry.EntityEntry
import dev.architectury.injectables.annotations.ExpectPlatform
import net.minecraft.world.item.BlockItem
import org.valkyrienskies.clockwork.content.curiosities.tools.gravitron.GravitronHandler
import org.valkyrienskies.clockwork.content.curiosities.tools.wanderwand.WanderwandHandler
import org.valkyrienskies.clockwork.content.kinetics.sequenced_seat.SequencedSeatEntity
import org.valkyrienskies.clockwork.content.physicalities.ballast.BallastBlockEntity
import org.valkyrienskies.clockwork.platform.api.network.PacketChannel
import java.util.function.BiConsumer

object SharedValues {
    @get:ExpectPlatform
    @JvmStatic
    val packetChannel: PacketChannel
        get() {
            throw AssertionError()
        }

    @ExpectPlatform
    @JvmStatic
    fun customRenderedRegisterer(): BiConsumer<CWItem, CustomRenderedItemModelRenderer> {
        throw AssertionError()
    }

    @ExpectPlatform
    @JvmStatic
    fun customBlockItemRenderedRegisterer(): BiConsumer<BlockItem, CustomRenderedItemModelRenderer> {
        throw AssertionError()
    }

    @get:ExpectPlatform
    @JvmStatic
    val sequencedSeat: EntityEntry<SequencedSeatEntity>
        get() {
            throw AssertionError()
        }

    @get:ExpectPlatform
    @JvmStatic
    val ballast: BlockEntityEntry<BallastBlockEntity>
        get() {
            throw AssertionError()
        }



    @get:ExpectPlatform
    @JvmStatic
    val gravitronHandler: GravitronHandler
        get() {
            throw AssertionError()
        }

    @get:ExpectPlatform
    @JvmStatic
    val wanderwandHandler: WanderwandHandler
        get() {
            throw AssertionError()
        }

    @get:ExpectPlatform
    @JvmStatic
    val auricHandler: WanderWandClusterRenderer
        get() {
            throw AssertionError()
        }
}